src/main/java/edu/harvard/iq/dataverse/engine/command/impl/UpdateMetadataBlockFacetRootCommand.java

```diff
@@ -0,0 +1,64 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.DataverseMetadataBlockFacet;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.engine.command.AbstractCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author adaybujeda
+ */
+@RequiredPermissions( Permission.EditDataverse )
+public class UpdateMetadataBlockFacetRootCommand extends AbstractCommand<Dataverse> {
+
+    private final Dataverse editedDv;
+    private final boolean metadataBlockFacetRoot;
+
+    public UpdateMetadataBlockFacetRootCommand(DataverseRequest aRequest, Dataverse editedDv, boolean metadataBlockFacetRoot) {
+        super(aRequest, editedDv);
+        this.editedDv = editedDv;
+        this.metadataBlockFacetRoot = metadataBlockFacetRoot;
+    }
+
+    @Override
+    public Dataverse execute(CommandContext ctxt) throws CommandException {
+        if(editedDv.isMetadataBlockFacetRoot() != metadataBlockFacetRoot) {
+            // Update metadata block facets when root changes value
+            // if you set root to be false (i.e. inherit), it should clear the blocks.
+            // if you set to true (i.e. use your own), it should make a copy of what is in the parent
+            List<DataverseMetadataBlockFacet> newBlockFacets = Collections.emptyList();
+            if (metadataBlockFacetRoot) {
+                newBlockFacets = editedDv.getMetadataBlockFacets().stream().map(blockFacet -> {
+                    DataverseMetadataBlockFacet metadataBlockFacet = new DataverseMetadataBlockFacet();
+                    metadataBlockFacet.setDataverse(editedDv);
+                    metadataBlockFacet.setMetadataBlock(blockFacet.getMetadataBlock());
+                    return metadataBlockFacet;
+                }).collect(Collectors.toList());
+            }
+            editedDv.setMetadataBlockFacets(newBlockFacets);
+        }
+
+        editedDv.setMetadataBlockFacetRoot(metadataBlockFacetRoot);
```

Comment:
@adaybujeda  very small thing (and it could be wrong), should these lines be in the if clause so they only get called when you "switch" - I can't recall if the save or anything here would update any "update times" or anything like that.

Reply:
I left the update outside the `if` to refresh the `modificationTime` (which is done on save) even if there is no change in the data.

Do you want me to change?

Reply:
I think that's my point - nothing is really getting modified, so let's change it. And acually one more related request then, in the API where you would call this command, first check to see if ti changed so that it doesn't call the command if it doesn't have to (so that it doesn't get logged)

Reply:
ok. Updating

Reply:
Update completed.
Updated as well the db script filename to: `V5.11.1.5__8536-metadata-block-facet.sql` as requested.